tests/general/MementoBearTest.py

```diff
@@ -0,0 +1,306 @@
+        self.section = Section('')
+        MementoBear.check_prerequisites = lambda *args: True
+        self.uut = MementoBear(self.section, Queue())
+
+    def test_run(self):
+        with requests_mock.Mocker() as m:
+            m.add_matcher(custom_matcher)
+
+            # Register archived links
+            memento_archive_status_mock(m, 'https://www.google.com')
+            memento_archive_status_mock(m, 'https://www.facebook.com')
+            memento_archive_status_mock(
+                m, 'https://github.com/mementoweb/py-memento-client/')
+
+            # Register not archived links
+            memento_archive_status_mock(
+                m, 'https://www.facebook.com/coala', False)
+
+            valid_file = """
+            # Archived links should not yields error
+            https://www.google.com
+            https://www.facebook.com
+            https://github.com/mementoweb/py-memento-client/
+
+            http://httpbin.org/status/401  # Unauthorized
+
+            # Templated URLs can't be checked
+            "http://httpbin.org/{status}/404".format(...)
+            "http://httpbin.org/$status/404"
+
+            # Not a link
+            http://not a link dot com
+            http://www.%s.com
+            http://www.%d.com
+            http://www.%f.com
+
+            # Example.com URLs should be ignored
+            http://sub.example.com/404
+            http://sub.example.com/something/404
+
+            # Invalid/dead URLs should be ignored
+            http://httpbin.org/status/404
+            http://httpbin.org/status/410
+            http://httpbin.org/status/500
+            http://httpbin.org/status/503
+            http://www.notexample.com/404
+            http://exampe.com/404
+            http://example.co.in/404
+            """.splitlines()
+
+            self.check_results(self.uut, valid_file, [])
+
+            # Not archived links should be detected by MementoBear
+            invalid_file = """
+            # Not archived links should yields error
+            https://www.facebook.com/coala
+            """.splitlines()
+
+            self.check_results(
+                self.uut,
+                invalid_file,
+                [Result.from_values(
+                    'MementoBear',
+                    ('This link is not archived yet, visit '
+                     'https://web.archive.org/save/https://www.facebook.com/co'
+                     'ala to get it archived.'),
+                    severity=RESULT_SEVERITY.INFO,
+                    line=3,
+                    file='default')],
+                filename='default')
+
+    def test_check_prerequisites(self):
+        with requests_mock.Mocker() as m:
+            m.add_matcher(custom_matcher)
+            self.assertTrue(MementoBear.check_prerequisites())
+
+    def test_multiple_results_per_line(self):
+        with requests_mock.Mocker() as m:
+            m.add_matcher(custom_matcher)
+
+            test_file = """
```

Comment:
done in https://github.com/coala/coala-bears/pull/1719/commits/f524628017c96a46dba6560e250bfeb295407636 by using `LocalBearTestHelper.check_line_result_count()` in line 220